htdocs/core/actions_sendmails.inc.php

```diff
@@ -434,10 +434,10 @@
 					setEventMessages($mesg, null, 'mesgs');
 
 					$moreparam = '';
-					if (isset($paramname2) || isset($paramval2)) {
+					if (isset($paramname2) && isset($paramval2)) {
```

Comment:
Why changing the || into &&  ?

If $paramval2 is set and not $paramname2, we should get '&mid='.$paramval2

Reply:
wit the `||`, you allow to call `$paramname2` even if it is not defined. This lead to a php error (version 19.0.0):
![screenshot](https://github.com/Dolibarr/dolibarr/assets/162124543/0f6a6805-58dd-4815-95e2-1cc7d373aaca)

We could also remove  `|| isset($paramval2)`

Reply:
So this means we must replace
`($paramname2 ? $paramname2 : 'mid')`
with
`(!empty($paramname2) ? $paramname2 : 'mid')`

Reply:
ok, if `empty($paramname2)`does not return an error when `$paramname2`is not set.
This is one of the main reasons I have a lot of crash with Dolibarr 19.0.0 with php 8.1. 
[https://wiki.php.net/rfc/undefined_property_error_promotion](php documentation)

Reply:
I replaced with `$paramname ?? 'id'`. I tested and it worked.